spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorProperties.java

```diff
@@ -90,11 +90,11 @@ public void setHost(String host) {
 	}
 
 	@NotNull
-	public Encoding getEncoder() {
+	public TcpConnectionFactoryProperties.Encoding getEncoder() {
```

Comment:
Looks like this `Encoding` is on wrong level.
Let's pull it to the top level class!

Reply:
+1